draft-kazuho-httpbis-priority.md

```diff
@@ -137,94 +137,32 @@ multiple experiments from independent research have shown that simpler schemes
-that receives a setting in violation of these requirements MUST treat it as a
-connection error of type PROTOCOL_ERROR for HTTP/2 {{!RFC7540}}, or of type
-H3_SETTINGS_ERROR for HTTP/3 {{!I-D.ietf-quic-http}}.
-
-In HTTP/2, the setting SHOULD appear in the first SETTINGS frame and peers
-MUST NOT process the setting if it is received multiple times in order to
-avoid changing the agreed upon prioritization scheme.
-
-If there is a prioritization scheme supported by both the client and server,
-then the server's preference order prevails and both peers SHOULD
-only use the agreed upon priority scheme for the remainder of the session.
-The server chooses because it is in the best position to know what
-information from the client is of the most value.
-
-Once the negotiation is complete, endpoints MAY stop sending hop-by-hop
-prioritization signals that were not negotiated in order to conserve bandwidth.
-However, endpoints SHOULD continue sending end-to-end signals (e.g., the
-Priority header field), as that might have meaningful effect to other nodes that
-handle the HTTP message.
-
-## Defined Prioritization Scheme Values
-
-This document defines two prioritization scheme values for use with the
-SETTINGS_PRIORITIES setting.
-
-### H2_TREE {#settings-h2-scheme}
-
-This document defines the priority scheme identifier H2_TREE (8-bit value of 1)
-that indicates support for HTTP/2-style priorities ({{!RFC7540}}, Section 5.3).
-
-The H2_TREE priority scheme identifier MUST NOT be be sent in an HTTP/3 settings
-because there is no defined mapping of this scheme. Endpoints MUST treat receipt
-of H2_TREE as a connection error of type H3_SETTINGS_ERROR.
-
-### URGENCY {#settings-this-scheme}
-
-This document defines the priority scheme identifier URGENCY (8-bit value of 2)
-that indicates support for the extensible priority scheme defined in the present
-document.
-
-An intermediary connecting to a backend server SHOULD declare support for the
-extensible priority scheme when and only when all the requests that are to be
-sent on that backend connection originates from one client-side connection that
-has negotiated the use of the extensible priority scheme (see {{fairness}}).
+## Disabling HTTP/2 Priorities
+
+The problems and insights set out above are motivation for allowing endpoints to
+opt out of using the HTTP/2 priority scheme, in favor of using an alternative
+such as the scheme defined in this specification. Endpoints would benefit from
+understanding their peer's intention, so the new
+SETTINGS_DEPRECATE_HTTP2_PRIORITIES is defined. The value of the parameter MUST
+be 0 or 1.
+
+Enpoints MUST send this SETTINGS parameter as part of the first SETTINGS frame.
+When the peer receives the first SETTINGS frame, it learns if the sender has
+deprecated the HTTP/2 priority scheme, by consulting the value of
+SETTINGS_DEPRECATE_HTTP2_PRIORITIES parameter (or through lack of that
```

Comment:
What does "(or through lack of that parameter)" mean, exactly?  I assume if it's not present, then it indicates HTTP/2 priorities are (at least possibly) supported?

Reply:
If a client and a server supports `SETTINGS_DEPRECATE_HTTP2_PRIORITIES` but a client omits the setting then the "initial value" of 0 is used (meaning that deprecate is not enabled). I also just added the requirement that once a setting is expressed it cannot be changed.